hadoop-tools/hadoop-sls/src/main/java/org/apache/hadoop/yarn/sls/AMRunner.java

```diff
@@ -53,7 +53,7 @@
 
 public class AMRunner {
   private static final Logger LOG = LoggerFactory.getLogger(AMRunner.class);
-  static int REMAINING_APPS = 0;
+  static int remainingApps = 0;
```

Comment:
Does this field need to be static? Spotbugs warns because this way it isn't threadsafe if multiple AMRunner instances write this. Referencing it via AMRunner.remaningApps should make the spotbugs warning disappear but it won't make this threadsafe.

Reply:
As the spotbugs warning is strictly related to this remainingApps field, I'd prefer to fix it with https://issues.apache.org/jira/browse/YARN-11102

Reply:
However if only one class is modifying this static variable a time (or if the owner class is a singleton) there is no problem with having a static field being modified from an instance method (a basic example would be to count the number of instances of a class in a completely single threaded environment).

Reply:
Indeed, this idiom has its places, the SpotBugs only warns us because it is generally not a good idea. All in all, in this case, the static integer is simply a poor architectural decision, therefore I advised in YARN-11102 to make it an instance member, thus eliminating all kinds of errors.

Reply:
Definitely agree.